build_tools/build_detect_platform

```diff
@@ -317,9 +317,8 @@ EOF
     # Test whether jemalloc is available
     if echo 'int main() {}' | $CXX $CFLAGS -x c++ - -o /dev/null -ljemalloc \
       2>/dev/null; then
-        PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -ljemalloc"
-        JAVA_LDFLAGS="$JAVA_LDFLAGS -ljemalloc"
         JEMALLOC=1
+        WITH_JEMALLOC_FLAG=1
```

Comment:
Just wondering if `JEMALLOC` itself could serve the purpose instead of adding another flag. 
Asking because both `JEMALLOC` and `WITH_JEMALLOC_FLAG` have no difference in their meaning to someone who has not spent enough time looking at the code.